planck-cljs/src/planck/http.cljs

```diff
@@ -109,19 +109,26 @@
   (fn [request]
     (client (assoc request :timeout (or (:timeout request) timeout)))))
 
+(defn- str->bytes [s]
+  (let [str->utf8  (comp js/unescape js/encodeURIComponent)
+        str->chars (fn [s] (mapv #(.charCodeAt %) s))]
+    (-> s str->utf8 str->chars)))
+
 (defn- generate-form-data [params]
   (conj (mapv (fn [[k v]]
                 (if (coll? v)
-                  (str content-disposition k "\"; filename=\"" (second v) "\"\n"
-                    "Content-Type: application/octet-stream\n\n"
-                    (first v))
-                  (str content-disposition k "\"\n\n" v))) params) "--\n"))
+                  (into (str->bytes (str content-disposition k "\"; filename=\"" (second v) "\"\n"
+                                      "Content-Type: application/octet-stream\n\n"))
+                    (cond-> (first v) (string? (first v)) str->bytes))
```

Comment:
This is a little pedantic but as the second argument to `into`, this should be aligned with the first argument above, right?

Reply:
Separately, is `cond->` being used here to return `nil` if `(first v)` is not a string? Just wondering if an `if` construct would be clearer (perhaps not, though—my experience is limited and happy to use this approach if it's pretty clear what it's doing).

Reply:
Oh, are you used to the arguments to `into` being aligned? As in the following?

```clojure
(into [1]
      [2 3 4])
```

The only bits of code I'm accustomed to aligning that way are `and` and `or`, and many times I've seen code involving `->` and `->>` all nicely vertically lined up. (FWIW, my style preferences have come from the ClojureScript codebase—I don't have strong feelings about them, other than alignment of `and` and `or` makes things easier to read, for me at least.

With respect to `cond->`, no, it is not returning `nil` if `(first v)` is not a string. It is really saying, if `(first v)` is a vector (of bytes), then it is fine, but if it is a string, then convert it using `str->bytes`.

Here are simple examples of `cond->` (neither return `nil`):

```
cljs.user=> (cond-> 1 true inc)
2
cljs.user=> (cond-> 1 false inc)
1
```

I suppose, writing `(cond-> (first v) (string? (first v)) str->bytes)` using an `if` would end up looking like:

```clojure
(if (string? (first v))
  (str->bytes (first v))
  (first v))
```

Happy to change it to this if that looks clearer. :)

Reply:
Ugh, I'm a dummy. I think I've been away too long in Rubyland.

1. Regarding the indentation: I think I saw the `let` structured that way and misremembered it as being the way all indentation worked. 

2. Regarding the return value of `cond->`: I don't know why I thought the result was `nil`. It wasn't particularly early here so I can't blame a lack of caffeine. My apologies!

3. Regarding the clarity of `cond->`: I suspect this is more on me than anything else. The `if` construct is clearer in some ways, but has the disadvantage of being duplicative. I would guess using `cond->` like this is pretty idiomatic, right?

Reply:
Yeah, using `cond->` with a single test and single function to mean "if this is true then do this additional little bit of processing to the thing" is probably the most common use of `cond->`, and it is used to eliminate the duplication.

TBH you rarely see `cond->` with multiple tests for some reason.